client/txn.go

```diff
@@ -71,5 +174,28 @@ func (t *Txn) Prepare(calls ...Call) {
 func (t *Txn) Flush() error {
 	calls := t.prepared
 	t.prepared = nil
+	if len(calls) == 0 {
+		return nil
+	}
 	return t.kv.Run(calls...)
 }
+
+func (t *Txn) updateNeedsEndTxn(calls []Call) {
+	for _, c := range calls {
+		if b, ok := c.Args.(*proto.BatchRequest); ok {
```

Comment:
I don't think see, though we're not disallowing it. I'm pretty sure `TxnCoordSender` wouldn't handle such a request properly.